go1.x/cookiecutter-aws-sam-eventbridge-hello-golang/{{cookiecutter.project_name}}/README.md

```diff
@@ -0,0 +1,146 @@
+
+```bash
+{{ cookiecutter.project_name }}$ sam build
+```
+
+The SAM CLI installs dependencies defined in `go.mod`, creates a deployment package, and saves it in the `.aws-sam/build` folder.
+
+Test a single function by invoking it directly with a test event. An event is a JSON document that represents the input that the function receives from the event source. Test events are included in the `events` folder in this project.
+
+Run functions locally and invoke them with the `sam local invoke` command.
+
+```bash
+{{ cookiecutter.project_name }}$ sam local invoke HelloWorldFunction --event events/event.json
+```
+
+The SAM CLI reads the application template to determine the EventBridge rule pattern and the functions that they invoke as a target. The `Events` property on each function's definition includes the source and detail-type of the types of events that will invoke the function.
+
+```yaml
+      Events:
+        HelloWorld:
+          Type: EventBridgeRule # More info about EventBridge Event Source: https://github.com/aws/serverless-application-model/blob/master/versions/2016-10-31.md#eventbridgerule
+          Properties:
+            Pattern:
+              source:
+                - aws.ec2
+              detail-type:
+                - EC2 Instance State-change Notification
+```
+
+## Add a resource to your application
+The application template uses AWS Serverless Application Model (AWS SAM) to define application resources. AWS SAM is an extension of AWS CloudFormation with a simpler syntax for configuring common serverless application resources such as functions, triggers, and APIs. For resources not included in [the SAM specification](https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md), you can use standard [AWS CloudFormation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-template-resource-type-ref.html) resource types.
+
+## Fetch, tail, and filter Lambda function logs
+
+To simplify troubleshooting, SAM CLI has a command called `sam logs`. `sam logs` lets you fetch logs generated by your deployed Lambda function from the command line. In addition to printing the logs on the terminal, this command has several nifty features to help you quickly find the bug.
+
+`NOTE`: This command works for all AWS Lambda functions; not just the ones you deploy using SAM.
+
+```bash
+{{ cookiecutter.project_name }}$ sam logs -n HelloWorldFunction --stack-name {{ cookiecutter.project_name }} --tail
+```
+
+You can find more information and examples about filtering Lambda function logs in the [SAM CLI Documentation](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-logging.html).
+
+## Unit tests
+
+We use `testing` package that is built-in in Golang and you can simply run the following commands to run the tests:
+
+```bash
+go test -v ./hello-world/
+```
+
+## Steps to End to End (E2E) test using the AWS Console
```

Comment:
Will keep that in mind for all future commits. Thanks

Reply:
I can remove the steps to E2E test in the console. The addition of that was a suggestion from @moelasmar from below

> Ok, I checked this internally. it is Ok to have this sample as it is as long as we include the steps to test this sample end to end in the read me file.

Reply:
I see. If we really think this is needed, we need to provide ways for customers to remove it. I prefer to have everything in the stack, as there shouldn't be a need for a customers to use `sam init` and then have to go to the console for things (not a great experience). Maybe EC2 is just a bad example and could use something else that would work with SAM Accelerate instead.

If it's a bigger change, I would be more inclined to remove this and find a way to integrate SAM Accelerate and expanding the example.

Reply:
I agree that the customer experience isn't the best having to manually End to End test using the console. We can remove it for now, and I'll create a SIM item for my team to look into a better way of do it (possibly using SAM Accelerate). Then we can expand it to all of the EventBridge app templates instead of just this one.